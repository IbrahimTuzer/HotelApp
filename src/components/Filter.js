import { StyleSheet, Text, View, ScrollView, Pressable } from 'react-native'
import React from 'react'

const Filter = () => {
  return (
    
      <ScrollView
       horizontal
       showsHorizontalScrollIndicator= {false}
       >
      <Pressable style={styles.filterText}>
      <Text style={styles.text}>Marmaris</Text>
      </Pressable>
      <Pressable style={styles.filterText}>
      <Text style={styles.text}>Bodrum</Text>
      </Pressable>
      <Pressable style={styles.filterText}>
      <Text style={styles.text}>Akyaka</Text>
      </Pressable>
      <Pressable style={styles.filterText}>
      <Text style={styles.text}>Antalya</Text>
      </Pressable>
      
      
      </ScrollView>
   
  )
}

export default Filter

const styles = StyleSheet.create({
    filterText:{
        flexDirection: "row",
        borderRadius: 20,
        width:82,
        height:32,
        alignItems:"center",
        justifyContent: "center",
        margin: 10,
        backgroundColor: "#D9D9D9"
    },
    
})