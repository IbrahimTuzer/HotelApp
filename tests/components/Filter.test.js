import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Filter from "../../src/components/Filter.js";

describe("Filter Component Tests", () => {
  const mockDistricts = ["District1", "District2", "District3"];

  it("renders correctly", () => {
    const { toJSON } = render(<Filter districts={mockDistricts} onSelectDistrict={() => {}} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders the correct number of Pressable elements", () => {
    const { getAllByTestId } = render(<Filter districts={mockDistricts} onSelectDistrict={() => {}} />);
    const pressableElements = getAllByTestId("pressable-element");
    expect(pressableElements).toHaveLength(mockDistricts.length);
  });

  it("calls onSelectDistrict with the correct district when Pressable is pressed", () => {
    const onSelectDistrictMock = jest.fn();
    const { getByText } = render(<Filter districts={mockDistricts} onSelectDistrict={onSelectDistrictMock} />);

    mockDistricts.forEach((district) => {
      const pressableElement = getByText(district);
      fireEvent.press(pressableElement);
      expect(onSelectDistrictMock).toHaveBeenCalledWith(district);
    });
  });
});
