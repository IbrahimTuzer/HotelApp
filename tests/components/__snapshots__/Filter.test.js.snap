// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component Tests renders correctly 1`] = `
<RCTScrollView
  horizontal={true}
  showsHorizontalScrollIndicator={false}
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#D9D9D9",
          "borderRadius": 20,
          "flexDirection": "row",
          "height": 32,
          "justifyContent": "center",
          "margin": 10,
          "width": 82,
        }
      }
    >
      <Text>
        District1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#D9D9D9",
          "borderRadius": 20,
          "flexDirection": "row",
          "height": 32,
          "justifyContent": "center",
          "margin": 10,
          "width": 82,
        }
      }
    >
      <Text>
        District2
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#D9D9D9",
          "borderRadius": 20,
          "flexDirection": "row",
          "height": 32,
          "justifyContent": "center",
          "margin": 10,
          "width": 82,
        }
      }
    >
      <Text>
        District3
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
